src/Sarif/Writers/SarifConsolidator.cs

```diff
@@ -0,0 +1,259 @@
+            result.Graphs = (this.RemoveGraphs ? null : result.Graphs);
+            result.GraphTraversals = (this.RemoveGraphTraversals ? null : result.GraphTraversals);
+            result.Stacks = (this.RemoveStacks ? null : result.Stacks);
+            result.WebRequest = (this.RemoveWebRequests ? null : result.WebRequest);
+            result.WebResponse = (this.RemoveWebResponses ? null : result.WebResponse);
+
+            // Truncate long Messages if configured
+            if (this.MessageLengthLimitBytes > 0)
+            {
+                if (result?.Message?.Text?.Length > this.MessageLengthLimitBytes)
+                {
+                    result.Message.Text = result.Message.Text.Substring(0, this.MessageLengthLimitBytes.Value);
+                }
+            }
+
+            // Trim Location and RelatedLocation properties
+            result.Locations = Trim(result.Locations);
+            result.RelatedLocations = Trim(result.RelatedLocations);
+
+            // Trim and Consolidate CodeFlows and ThreadFlows
+            ConsolidateCodeFlows(result);
+        }
+
+        public void ConsolidateCodeFlows(Result result)
+        {
+            if (result.CodeFlows != null)
+            {
+                foreach (CodeFlow cFlow in result.CodeFlows)
+                {
+                    foreach (ThreadFlow tFlow in cFlow.ThreadFlows)
+                    {
+                        Consolidate(tFlow);
+                    }
+                }
+            }
+        }
+
+        public void Consolidate(ThreadFlow tFlow)
+        {
+            if (tFlow.Locations != null)
+            {
+                TotalThreadFlowLocations += tFlow.Locations.Count;
+
+                for (int i = 0; i < tFlow.Locations.Count; ++i)
+                {
+                    ThreadFlowLocation tfl = tFlow.Locations[i];
+                    Trim(tfl.Location);
+
+                    if (tfl.Index == -1)
+                    {
+                        if (!_uniqueThreadFlows.TryGetValue(tfl, out int tflIndex))
+                        {
+                            tflIndex = _run.ThreadFlowLocations.Count;
+                            _run.ThreadFlowLocations.Add(tfl);
+                            _uniqueThreadFlows[tfl] = tflIndex;
+                        }
+
+                        tFlow.Locations[i] = new ThreadFlowLocation() { Index = tflIndex };
+                    }
+                }
+            }
+        }
+
+        public IList<Location> Trim(IList<Location> locations)
+        {
+            if (locations == null || locations.Count == 0) { return locations; }
+
+            HashSet<Location> uniqueLocations = new HashSet<Location>(Location.ValueComparer);
+            List<Location> newLocations = new List<Location>();
+
+            this.TotalLocations += locations.Count;
+            for (int i = 0; i < locations.Count; ++i)
+            {
+                Location loc = locations[i];
```

Comment:
>loc [](start = 25, length = 3)

Can we use 'location' as a var name here? Generally, we tend to favor verbose (arguably over verbose) names in this code base.

Reply:
Changed. The victim of repeated refactors in the initial authoring.